src/core/ext/xds/xds_client_grpc.cc

```diff
@@ -178,6 +184,29 @@ absl::StatusOr<RefCountedPtr<GrpcXdsClient>> GrpcXdsClient::GetOrCreate(
   return xds_client;
 }
 
+grpc_slice GrpcXdsClient::DumpAllClientConfigs()
+    ABSL_NO_THREAD_SAFETY_ANALYSIS {
+  auto xds_client =
+      GrpcXdsClient::GetOrCreate(ChannelArgs(), "grpc_dump_xds_configs()");
```

Comment:
There's not much point in creating a new `XdsClient` if we don't already have one.  I suggest doing something like this:

```
RefCountedPtr<GrpcXdsClient> xds_client;
{
  MutexLock lock(g_mu);
  xds_client =
      g_xds_client->RefIfNonZero(DEBUG_LOCATION, "csds")
          .TakeAsSubclass<GrpcXdsClient>();
}
```

Then just check if `xds_client` is not null.